usi/src/main/java/daq/usi/UsiImpl.java

```diff
@@ -15,39 +13,42 @@
 import java.util.Map;
 
 public class UsiImpl extends USIServiceGrpc.USIServiceImplBase {
-  private final Map<String, BaseSwitchController> switchControllers;
+  private final Map<String, SwitchController> switchControllers;
 
   public UsiImpl() {
     super();
     switchControllers = new HashMap<>();
   }
 
   private SwitchController getSwitchController(SwitchInfo switchInfo) {
-    if (switchInfo.getModel().equals(OVS_SWITCH)) {
-      return new OpenVSwitch();
-    }
-    String repr = String.join(",", switchInfo.getModel().toString(), switchInfo.getIpAddr(),
-        String.valueOf(switchInfo.getCmdTelnetPort()), switchInfo.getUsername(),
+    String repr = String.join(",", switchInfo.getModel().toString(),
+        switchInfo.getIpAddr(), switchInfo.getUsername(),
         switchInfo.getPassword());
-    BaseSwitchController sc = switchControllers.get(repr);
-    if (sc == null) {
+    SwitchController sc = switchControllers.computeIfAbsent(repr, key -> {
+      final SwitchController newController;
```

Comment:
extract switch statement to function (createNewController or something like that) - lengthy embedded functions aren't very readable...